### ISSUE TITLE
FreeBSD support

### ISSUE THREAD

**Issue by blkt** [jni-rs/jni-rs#289]
Hello everyone!

while trying to compile some personal stuff under FreeBSD, I found it was relatively easy to add support to FreeBSD to `jni-rs`.
I modified the `build.rs` file this way to make it work, but my codebase is far from being a real world use case of JNI.
```
diff --git i/build.rs w/build.rs
index 268df387d251..584caa6f8cc3 100644
--- i/build.rs
+++ w/build.rs
@@ -20,7 +20,7 @@ use std::{

 #[cfg(target_os = "windows")]
 const EXPECTED_JVM_FILENAME: &str = "jvm.dll";
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
 const EXPECTED_JVM_FILENAME: &str = "libjvm.so";
 #[cfg(target_os = "macos")]
 const EXPECTED_JVM_FILENAME: &str = "libjli.dylib";
```

I ran the tests on both a docker container and on FreeBSD and they give the same results.
https://gist.github.com/blkt/758c57f3ffa91085e0c3aa4f539b1ba9

is there any other way I could ensure it's working as expected?
Would you accept such a simple PR?

Regards

**Comment by dmitry-timofeev** [comment]
Hi @blkt , 

Looks reasonable — if a JVM works on FreeBSD, then jni-rs must work too. Will happily merge the PR. 

I see in the output that the integration tests didn't run, only a couple of unit tests. To test the project works with a JVM, please pass `--features=invocation` when testing. See the Contributing guide for details.

**Comment by maurolacy** [comment]
Unfortunately [Github Actions don't support FreeBSD](https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners#supported-runners-and-hardware-resources).

There's a [freebsd-vm](https://github.com/marketplace/actions/freebsd-vm) for it as an alternative, though.

@blkt, I think this would be a nice addition to your PR. ;-)

In any case, I'll create an issue to investigate this.

**Comment by blkt** [comment]
Hello @dmitry-timofeev, @maurolacy,

thank for your prompt reply, I'll get it done as quickly as I can.

**Comment by dmitry-timofeev** [comment]
If it works locally with `invocation` feature, and it's non-trivial to add tests on FreeBSD, please don't spend too much time on it, I think we can skip it for now (if there is an OpenJDK build for FreeBSD, it must be tested).

**Comment by blkt** [comment]
Hello everyone, just a small update. First, I'm sorry for the long wait, I finally found some time to work on it. Today I wrestled a bit with GitHub Actions for FreeBSD as the [freebsd-vm](https://github.com/marketplace/actions/freebsd-vm) plugin runs VirtualBox on a macOS host, so commands seem to be performed on the host macOS instead of the guest FreeBSD.

I'll open the PR when I get that working.

Last but not least, happy new year everyone!
